lib/internal/main/check_syntax.js

```diff
@@ -60,7 +60,7 @@ function loadESMIfNeeded(cb) {
 async function checkSyntax(source, filename) {
   let isModule = true;
   if (filename === '[stdin]' || filename === '[eval]') {
-    isModule = getOptionValue('--input-type') === 'module';
+    isModule = getOptionValue('--input-type') === 'module' || getOptionValue('--experimental-type') === 'module';
```

Comment:
I think this condition won't give the right result for `node --experimental-type=module --input-type=commonjs`.

Reply:
An error is thrown if the flags are used together.

Reply:
Ah, that should be documented, then. When I read "String input provided via `--eval` or STDIN, if `--input-type` is unspecified.", the first thing I wondered is what happens when both are specified (and saw this change).

Reply:
I updated this to allow them to be used together, and for `--input-type` to take precedence; and I added a test.

Side note, why does `--input-type` affect `--check`? `--check` is meant for files, but `--input-type` is only for non-file input. This seems like a bug that we should fix separately.